cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -296,6 +297,15 @@ Map<ProgramRunId, RunRecordDetail> getRuns(ApplicationId applicationId, ProgramR
    */
   void addApplication(ApplicationId id, ApplicationSpecification specification);
 
+  /**
+   * Creates new application if it doesn't exist. Updates existing one otherwise.
+   *
+   * @param id            application id
+   * @param specification application specification to store
+   * @param owner         user name (owner) of the application version
+   */
+  void addApplication(ApplicationId id, ApplicationSpecification specification, @Nullable String owner);
```

Comment:
is this an owner, or the user who made the change? Owner is normally used to describe permissions. It's better to name it 'author' if it's just the person who made the edit.

Though this is one benefit of including more information in the change summary, you wouldn't need to add another method.

Reply:
This is the author - we can rename this. I added these new methods so it won't break existing tests - although I agree those tests can be refactored